### PR TITLE
8329766: Serial: Refactor SerialBlockOffsetTable API

### PR DESCRIPTION
**Pull request by albertnetymk**
Simple refactoring towards unifying BOT API between Serial and Parallel.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8329766](https://bugs.openjdk.org/browse/JDK-8329766): Serial: Refactor SerialBlockOffsetTable API (**Enhancement** - P4)

### Reviewers
 * [Guoxiong Li](https://openjdk.org/census#gli) (@lgxbslgx - Committer)
 * [Ivan Walulya](https://openjdk.org/census#iwalulya) (@walulyai - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18651/head:pull/18651` \
`$ git checkout pull/18651`

Update a local copy of the PR: \
`$ git checkout pull/18651` \
`$ git pull https://git.openjdk.org/jdk.git pull/18651/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18651`

View PR using the GUI difftool: \
`$ git pr show -t 18651`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18651.diff">https://git.openjdk.org/jdk/pull/18651.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18651#issuecomment-2039565028)

### CODE REVIEW

**Review by lgxbslgx — APPROVED** [review]
Looks good.

File: src/hotspot/share/gc/serial/serialBlockOffsetTable.cpp [hunk 1]
```diff
@@ -77,6 +77,11 @@ void SerialBlockOffsetTable::resize(size_t new_word_size) {
   }
 }
 
+static void fill_range(uint8_t* start, uint8_t* end, uint8_t v) {
+  // + 1 for inclusive
```
Comment (walulyai) [inline src/hotspot/share/gc/serial/serialBlockOffsetTable.cpp:81]:
```suggestion
  // + 1 for inclusive.
```

File: src/hotspot/share/gc/serial/serialBlockOffsetTable.hpp [hunk 2]
```diff
@@ -66,6 +62,12 @@ class SerialBlockOffsetTable: public CHeapObj<mtGC> {
     return ReservedSpace::allocation_align_size_up(number_of_slots);
   }
 
+  // Mapping from address to object start array entry
```
Comment (walulyai) [inline src/hotspot/share/gc/serial/serialBlockOffsetTable.hpp:65]:
```suggestion
  // Mapping from address to object start array entry.
```

File: src/hotspot/share/gc/serial/serialBlockOffsetTable.hpp [hunk 3]
```diff
@@ -66,6 +62,12 @@ class SerialBlockOffsetTable: public CHeapObj<mtGC> {
...
+  uint8_t* entry_for_addr(const void* const p) const;
+
+  // Mapping from object start array entry to address of first word
```
Comment (walulyai) [inline src/hotspot/share/gc/serial/serialBlockOffsetTable.hpp:68]:
```suggestion
  // Mapping from object start array entry to address of first word.
```

File: src/hotspot/share/gc/serial/serialBlockOffsetTable.cpp [hunk 4]
```diff
@@ -77,6 +77,11 @@ void SerialBlockOffsetTable::resize(size_t new_word_size) {
   }
 }
 
+static void fill_range(uint8_t* start, uint8_t* end, uint8_t v) {
```
[the remark this answers is not in the document]
Comment (walulyai) [reply src/hotspot/share/gc/serial/serialBlockOffsetTable.cpp:80]: I suggest we write out the `v` as `value`, I don't see what we save by the abbreviation.

**Review by walulyai — CHANGES_REQUESTED** [review]

**Review by lgxbslgx — APPROVED** [review]

**Review by walulyai — APPROVED** [review]

**Comment by albertnetymk** [comment]
Thanks for review.

/integrate

**Pull request closed without merging**